### PR TITLE
Merge feature/dash into dev

### PR DESCRIPTION
**Pull request by bludnic**

### CODE REVIEW

File: src/lib/bitcoin/btc-base-api.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+  /**
+   * Creates a raw DOGE transaction as a hex string.
+   * @param {string} address target address
+   * @param {number} amount amount to send
+   * @param {Array<{txid: string, amount: number, vout: number}>} unspents unspent transaction to use as inputs
+   * @returns {string}
+   */
+  _buildTransaction (address, amount, unspents) {
+    amount = new BigNumber(amount).times(this.multiplier).toNumber()
```
Comment (MaaKut) [inline src/lib/bitcoin/btc-base-api.js]: Apply `Math.floor` here

File: src/lib/bitcoin/btc-base-api.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    const txb = new bitcoin.TransactionBuilder(this._network)
+    txb.setVersion(1)
+
+    const target = amount + new BigNumber(this.getFee()).times(this.multiplier).toNumber()
```
Comment (MaaKut) [inline src/lib/bitcoin/btc-base-api.js]: `Math.floor`

File: src/i18n/de.json [hunk 3]
```diff
@@ -33,10 +33,9 @@
     "scan_recipient_button": "Scan",
     "send_adm": "Send ADM",
```
Comment (zyuhel) [inline src/i18n/de.json]: send_adm/send_bnb should be removed as well

File: src/components/transactions/BtcTransaction.vue [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  },
+  computed: {
+    cryptoKey () {
+      return this.crypto.toLowerCase()
+    },
+    transaction () {
+      return this.$store.getters[`${this.cryptoKey}/transaction`](this.id) || { }
+    },
+    amount () {
+      if (!this.transaction.amount) return ''
```
Comment (MaaKut) [inline src/components/transactions/BtcTransaction.vue]: ```if (!isFinite(this.transaction.amont))...``` (from `master`)

File: src/lib/bitcoin/btc-base-api.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+    const recipients = getUnique(tx.vout.reduce((list, out) => {
+      list.push(...out.scriptPubKey.addresses)
+      return list
+    }, []))
+
+    if (direction === 'from') {
+      // Disregard our address for the outgoing transaction
+      const idx = recipients.indexOf(this._address)
+      if (idx >= 0) recipients.splice(idx, 1)
```
Comment (MaaKut) [inline src/lib/bitcoin/btc-base-api.js]: Do this only if `recipients.length > 1` (see https://github.com/Adamant-im/adamant-im/pull/237/files)

File: src/components/transactions/BtcTransaction.vue [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      const partners = this.$store.state.partners
+      Object.keys(partners).some(uid => {
+        const partner = partners[uid]
+        if (partner[this.crypto] === address) {
+          admAddress = uid
+        }
+        return !!admAddress
+      })
+
+      if (!admAddress) {
```
Comment (zyuhel) [inline src/components/transactions/BtcTransaction.vue]: What is the situation then we doesn't have address in partners but we have it in messages?

**Comment by bludnic** [comment]
> `components/transactions/DogeTransaction.vue` and `lib/doge-api.js` can be removed

`lib/doge-api.js` is imported here. How to fix?
https://github.com/Adamant-im/adamant-im/blob/dev-dash/src/store/modules/doge/doge-getters.js#L1

File: src/lib/bitcoin/btc-base-api.js (hunk 1 above)
Comment (bludnic) [reply src/lib/bitcoin/btc-base-api.js]:
```js
amount = new BigNumber(amount).times(this.multiplier).toNumber()
amount = Math.floor(amount)
```

right?

File: src/lib/bitcoin/btc-base-api.js (hunk 2 above)
Comment (bludnic) [reply src/lib/bitcoin/btc-base-api.js]:
```js
const target = Math.floor(amount) + new BigNumber(this.getFee()).times(this.multiplier).toNumber()
```

right?

File: src/lib/bitcoin/btc-base-api.js (hunk 1 above)
Comment (MaaKut) [reply src/lib/bitcoin/btc-base-api.js]: yep

File: src/lib/bitcoin/btc-base-api.js (hunk 2 above)
Comment (MaaKut) [reply src/lib/bitcoin/btc-base-api.js]: yes

**Pull request merged**